src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+                    {
+                        // The parameter with keyword params takes an array type, then it cannot be matched more than once
+                        if (matchedTypes[parameterIndex]) return false;
+                        matchedTypes[parameterIndex] = true;
+                        paramsMatchedByArray = true;
+                    }
+                    else if (parameters[parameterIndex].IsParams
+                             && parameters.Last().Type is IArrayTypeSymbol paramsType
+                             && semanticModel.Compilation.ClassifyCommonConversion(argType.Type, paramsType.ElementType).Exists)
+                    {
+                        // The parameter with keyword params takes a variable number of arguments, compare its element type with the argument's type.
+                        if (matchedTypes[parameterIndex] && paramsMatchedByArray) return false;
+                        matchedTypes[parameterIndex] = true;
+                        paramsMatchedByArray = false;
+                    }
+                    else return false;
+
+                    if (targetArgument.Equals(arguments[i])) targetParamIndex = parameterIndex;
+                }
+                else return false;
+            }
+
+            // mark all optional parameters as matched
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                if (parameters[i].IsOptional || parameters[i].IsParams)
+                {
+                    matchedTypes[i] = true;
+                }
+            }
+
+            return Array.TrueForAll(matchedTypes, (item => item));
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var targetNodes = diagnostics.SelectAsArray(
+                d => root.FindNode(d.Location.SourceSpan, getInnermostNodeForTie: true)
+                         .GetAncestorsOrThis<ExpressionSyntax>().FirstOrDefault());
+
+            await editor.ApplyExpressionLevelSemanticEditsAsync(
+                document, targetNodes,
+                (semanticModel, targetNode) => true,
+                (semanticModel, currentRoot, targetNode) =>
+                {
+                    // All diagnostics have the same error code
+                    if (TryGetTargetTypeInfo(semanticModel, diagnostics[0].Id, targetNode, cancellationToken, out var nodeType, out var potentialConversionTypes)
+                        && potentialConversionTypes.Length == 1)
+                    {
+                        return ApplyFix(currentRoot, targetNode, potentialConversionTypes[0]);
+                    }
+
+                    return currentRoot;
```

Comment:
this seems not right (andreally bad).  this should be `return targetNode` instead.

Reply:
>  Can I know why here return targetNode is also okay

Oh, i might have gotten this wrong.  `return currentRoot` is likely the right thing, i think i flipped myself.